src/main/scala/nl/knaw/dans/api/sword2/DepositHandler.scala

```diff
@@ -81,28 +80,26 @@ object DepositHandler {
 
   def finalizeDeposit(mimeType: String)(implicit settings: Settings, id: String): Try[Unit] = {
     log.info(s"[$id] Finalizing deposit")
-    // TODO pass on the combination object of baseDir and baseURL
-    implicit val baseDir: File = new File(settings.bagStoreBaseDir)
-    implicit val baseUrl: URI  = new URI(settings.bagStoreBaseUri)
+    implicit val optBagStoreSettings: Option[BagStoreSettings] = settings.bagStoreSettings
     val tempDir = new File(settings.tempDir, id)
 
     val result = for {
-      _        <- checkBagStoreBaseDir()
+      _        <- optBagStoreSettings.map(checkBagStoreBaseDir).getOrElse(Success(()))
```

Comment:
- I do want the service to refuse to start if preconditions for normal functioning are not satisfied.
- If for some reason the system gets messed up *while* it is running, there will of course be an error when trying to access the non-existent directory. The question is: do we have to check above preconditions before *every* request? That seems a bit much for something that is very unlikely to happen.